scripting/src/main/java/org/infinispan/scripting/impl/ScriptingManagerImpl.java

```diff
@@ -165,10 +165,13 @@ public boolean containsScript(String taskName) {
          .orElseGet(() -> new SimpleBindings());
 
       SimpleBindings systemBindings = new SimpleBindings();
-      systemBindings.put(SystemBindings.CACHE_MANAGER.toString(), cacheManager);
+      DataTypedCacheManager cm = new DataTypedCacheManager(metadata.dataType(), context.getMarshaller(), cacheManager);
+      systemBindings.put(SystemBindings.CACHE_MANAGER.toString(), cm);
       systemBindings.put(SystemBindings.SCRIPTING_MANAGER.toString(), this);
       context.getCache().ifPresent(cache -> {
-         systemBindings.put(SystemBindings.CACHE.toString(), cache);
+         Cache<?, ?> c = cm.getCacheConfiguration(cache.getName()).compatibility().enabled()
```

Comment:
Aside from the fact that this should be fixed by protecting the other entry point as well, cm.getCacheConfiguration() doesn't probably do what you want since you can start a cache using a configuration with a different name